proto/baremetal/v1/host_pool_type.proto

```diff
@@ -0,0 +1,180 @@
+//
+// Copyright (c) 2025 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+
+syntax = "proto3";
+
+package baremetal.v1;
+
+import "google/protobuf/timestamp.proto";
+import "shared/v1/condition_status_type.proto";
+import "shared/v1/metadata_type.proto";
+
+// Contains the details of the host pool.
+//
+// The `spec` contains the desired details, and may be modified by the user. The `status` contains the current status of
+// the host pool, is provided by the system and can't be modified by the user.
+message HostPool {
+  // Unique identifier of the host pool.
+  string id = 1;
+
+  shared.v1.Metadata metadata = 2;
+  HostPoolSpec spec = 3;
+  HostPoolStatus status = 4;
+}
+
+// The spec contains the details of a host pool as desired by the user.
+message HostPoolSpec {
+  // Desired nodes
+  repeated HostPoolNodeSet node_sets = 1;
```

Comment:
For clusters this is a map instead of an array. The key of the map is the identifier of the node set. There are different opinions about that. I won't object to change the aproach, but then it should be everywhere, not just for new API types.